pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/logicaltable/BaseLogicalTableIntegrationTest.java

```diff
@@ -158,11 +171,70 @@ public void tearDown()
     cleanup();
   }
 
-  protected abstract List<String> getOfflineTableNames();
+  protected Map<String, List<File>> getOfflineTableDataFiles() {
```

Comment:
Should these functions be abstract ? If a test does not override one of them then there there will be duplicate data in realtime and offline tables ? 

The way these functions are used confused me until I read the code of all the tests

Reply:
No. There would not be any data duplication as specific tests override the specific methods. I did not make them abstract as to reduce the code duplication.